### DISCUSSION TITLE
BC04P and diff UUIDs, espresence doesn't see difference with same mac id.

### DISCUSSION THREAD

**Discussion by markheitz** [ESPresense/ESPresense#1173] [Q&A]
I have a blue charb beacon BC04P which is meant to send off a different UUID with motion.  I have it successfully set up in espresence for base functionality.  I can't seem to figure out how to have espresence recognize the difference in UUID.  It seems to go by MAC ID and not uuid.  IS there a way to set it to look like 2 diifferent devices.  xxxxxxf4 and xxxxxxf5, each with the same mac ID?   ESPresence looks to see the different UUID and will only report the difference with a hard restart and will stay with that UUID even if it goes back to original.  Appreciate any help.  

(I'm attaching this to my trash can.   using distance to remind me to take it out, and would be awesome if i could get the vibration trigger when trash man has come)  

UUID with F4
iBeacon:1b6295d5-4f74-4c58-a2d8-cd83ca26bdf4-3838-4949
office = {"id":"iBeacon:1b6295d5-4f74-4c58-a2d8-cd83ca26bdf4-3838-4949","idType":180,"rssi@1m":-59,"rssi":-71,"raw":2.20,"distance":1.21,"speed":0.06,"mac":"dd60030000ef","interval":2622}

UUID with F5
iBeacon:1b6295d5-4f74-4c58-a2d8-cd83ca26bdf5-3838-4949
office = {"id":"iBeacon:1b6295d5-4f74-4c58-a2d8-cd83ca26bdf5-3838-4949","idType":180,"rssi@1m":-59,"rssi":-67,"raw":1.69,"distance":1.34,"speed":0.02,"mac":"dd60030000ef","interval":3391}

**Comment by ih8gates** [comment]
Came here looking for this as well! Seems like ESPresense is smart enough to know it's one device and I need to make it less smart and see those UUIDs as separate devices.

**Comment by Terastar-Paperclip** [comment]
Tech Lead here — traced the actual cause, it's not that ESPresense goes by MAC instead of UUID (your own pasted JSON shows the `id` field really does change to the new UUID). It's `BleFingerprint::setId()`'s anti-flap logic: once a fingerprint locks onto an idType, a new id of the *same* idType gets rejected outright, even if the concrete id string is different. Both your UUIDs resolve to `ID_TYPE_IBEACON`, so the second one never gets through until a restart clears the lock.

This is deliberate anti-flap protection, not a bug in the general case — but it's blocking a legitimate use case here. Filed #2492 with the exact lines and a proposed scoped fix (let a same-idType id through when the id string itself differs, still block strictly-lower-priority types). It's a narrow diff but needs real verification against other idTypes before merging, so I'm not going to rush it in from a Q&A thread — marked **help wanted** if someone wants to pick it up and test it properly.
